### PR TITLE
add default chainid for mobile browsers 

### PR DESCRIPTION
**Pull request by pRoy24**
Fixes the mobile display for UI

### CODE REVIEW

File: app/src/components/main/index.tsx [hunk 1]
```diff
@@ -29,23 +29,28 @@ import { MarketWizardCreatorContainer } from '../market/sections/market_create/m
...
-  windowObj.ethereum.on('chainChanged', (chainId: string) => {
-    setNetworkId(chainId)
-    if (location.host === MAINNET_LOCATION && XDAI_NETWORKS.includes(chainId)) {
-      location.assign(`http://${XDAI_LOCATION}`)
-    }
-    if (location.host === XDAI_LOCATION && MAIN_NETWORKS.includes(chainId)) {
-      location.assign(`http://${MAINNET_LOCATION}`)
-    }
-  })
+  if (windowObj.ethereum) {
```
Comment (pRoy24) [inline app/src/components/main/index.tsx:43]: The default is only for the init case when there is no chain id.
Once the user logs in via a wallet, then windowObj.ethereum will be defined and this onChange handler would work as usual.

**Comment by pRoy24** [comment]
Fixes https://github.com/protofire/omen-exchange/issues/1640

File: app/src/components/main/index.tsx [hunk 2]
```diff
@@ -29,23 +29,28 @@ import { MarketWizardCreatorContainer } from '../market/sections/market_create/m
...
   const context = useWeb3Context()
   const windowObj: any = window
-
-  const [networkId, setNetworkId] = useState(windowObj.ethereum.chainId)
+  const host = window.location.hostname
+  let defaultChainID = 1
+  if (host === XDAI_LOCATION) {
+    defaultChainID = 100
+  }
+  const [networkId, setNetworkId] = useState(windowObj.ethereum ? windowObj.ethereum.chainId : defaultChainID)
```
Comment (Mi-Lan) [inline app/src/components/main/index.tsx:40]: One question why are we using defaultChainId variable and then networkId useState hook...why not just consolidate it to use hook and use setNetworkId to set default ones?

**Review by KadenZipfel — APPROVED** [review]

**Comment by pimato** [comment]
IMG_FD00C7D2BF7B-1

tried it on my mobile and it does not seem to work.

**Review by pimato — APPROVED** [review]

File: app/src/components/main/index.tsx (hunk 2 above)
Comment (pRoy24) [reply app/src/components/main/index.tsx:40]: NetworkID is not the same as chainid, they have the same values but they are not the same entity.
https://ethereum.stackexchange.com/questions/37533/what-is-a-chainid-in-ethereum-how-is-it-different-than-networkid-and-how-is-it

**Review by hexyls — APPROVED** [review]

**Pull request merged**